app/assets/javascripts/plot/export-csv.coffee

```diff
@@ -0,0 +1,309 @@
+    options = (@options.exporting or {}).csv or {}
+    itemDelimiter = options.itemDelimiter or ','
+    lineDelimiter = options.lineDelimiter or '\n'
+    # '\n' isn't working with the js csv data extraction
+    # Transform the rows to CSV
+    each(rows, (row, i) ->
+      val = ''
+      j = row.length
+      n = if useLocalDecimalPoint then 1.1.toLocaleString()[1] else '.'
+      while j--
+        val = row[j]
+        if typeof val == 'string'
+          val = '"' + val + '"'
+        if typeof val == 'number'
+          if n == ','
+            val = val.toString().replace('.', ',')
+        row[j] = val
+      # Add the values
+      csv += row.join(itemDelimiter)
+      # Add the line delimiter
+      if i < rows.length - 1
+        csv += lineDelimiter
+      return
+    )
+    csv
+
+  ###*
+  # Call this on click of 'Download CSV' button
+  ###
+
+  Highcharts.Chart::downloadCSV = ->
+    csv = @getCSV(true)
+    getContent(this, 'data:text/csv,\ufeff' + encodeURIComponent(csv), 'csv', csv, 'text/csv')
+    return
+
+  ###*
+  # Call this on click of 'Download XLS' button
+  ###
+
+  Highcharts.Chart::downloadXLS = ->
+    uri = 'data:application/vnd.ms-excel;base64,'
+    template = '<html xmlns:o="urn:schemas-microsoft-com:office:office"' +
+    'xmlns:x="urn:schemas-microsoft-com:office:excel" xmlns="http://www.w3.org/TR/REC-html40">' +
+    '<head><!--[if gte mso 9]><xml><x:ExcelWorkbook><x:ExcelWorksheets><x:ExcelWorksheet>' +
+    '<x:Name>Ark1</x:Name>' +
+    '<x:WorksheetOptions><x:DisplayGridlines/></x:WorksheetOptions></x:ExcelWorksheet>' +
+    '</x:ExcelWorksheets></x:ExcelWorkbook></xml><![endif]-->' +
+    '<style>td{border:none;font-family: Calibri, sans-serif;} .number{mso-number-format:"0.00";}</style>' +
+    '<meta name=ProgId content=Excel.Sheet>' + '<meta charset=UTF-8>' +
+    '</head><body>' + @getTable(true) + '</body></html>'
+
+    base64 = (s) ->
+      window.btoa(unescape(encodeURIComponent(s)))
+      # #50
+
+    getContent(this, uri + base64(template), 'xls', template, 'application/vnd.ms-excel')
+    return
+
+
+  # Add "Download CSV" to the exporting menu. Use download attribute if supported, else
+  # run a simple PHP script that returns a file. The source code for the PHP script can be viewed at
+  # https://raw.github.com/highslide-software/highcharts.com/master/studies/csv-export/csv.php
+  if Highcharts.getOptions().exporting
+    Highcharts.getOptions().exporting.buttons.contextButton.menuItems.push({
+      textKey: 'downloadCSV'
+      onclick: ->
+        @downloadCSV()
+        return
+
+    }, {
+      textKey: 'downloadXLS'
+      onclick: ->
+        @downloadXLS()
+        return
+    })
+  # Series specific
+  if seriesTypes.map
+    seriesTypes.map::exportKey = 'name'
+  if seriesTypes.mapbubble
+    seriesTypes.mapbubble::exportKey = 'name'
+  if seriesTypes.treemap
+    seriesTypes.treemap::exportKey = 'name'
+  return
+)
+# ---
+# generated by js2coffee 2.2.0
```

Comment:
Functionally, this PR is awesome!  My only complaints are about how we're going about it.  Namely, instead of dropping an outside file into the repo, our standard way of dealing with this sort of thing—for a number of reasons that I can go into more at length, if you want—is to depend on the external file through our package management system.

According to webjars.org , we can depend on this file in SBT by adding the entry `"org.webjars.bower" % "github-com-highcharts-export-csv" % "1.4.3"` to `build.sbt`.  0a5b2b73e6ebdd8a5350bdc9d6d86cb8f841429a is an example of a commit where something similar was done, in that we added a dependency to `build.sbt` and then specified that it should be included in NLW by adding lines to `simulation.scala.html` and `CompilerService.scala` to include the file by its URL.

With WebJars, it's sometimes a bit of a challenge to guess what URL we'll need to provide to get this file out of the WebJar.  The WebJars site says that the file is at `META-INF/resources/webjars/github-com-highcharts-export-csv/1.4.3/export-csv.js`.  As a result, I would guess that the URL we should use try to get it from is `lib/github-com-highcharts-export-csv/export-csv.js`.  Try that and see if it works.